Geocoded Peacekeeping Operations (Geo-PKO)

krihelskog opened the pull request:
Added read function to read the data. 
A sum variable 
A count variable for each year in a raster stack.

kvelleby commented:
Dear Kristine,

Thank you for this contribution. Here is a few comments I would like you to address before I can accept the pull request:

1. I want you to remove the `data_eurostat...R` files and send a separate pull request for these. I do not comment on this code below.
2. It should not be a newline between the function documentation and the function.
3. You need to compile the markdown for documentation using `devtools::document()`.
4. `gen_geopko_sum()` currently returns the name of the raster (you need to add a new line with `return(r)` or just `r`
5. Your two gen_ functions are currently using `terra::rasterize`, but we want to use `exactextractr` because it is more accurate. See `data_naturalearth.R` `gen_natural_earth_cover_share()` for an example using `exactextractr`.
6. After swapping to using exactextractr, check: `df <- read_geopko(); df |> dplyr::group_by(year) |> dplyr::summarize(n = dplyr::n()); r <- gen_geopko_count(); sum(values(r["1994"]), na.rm = T)` I.e., there should be the same number of events per year in the raster as in the vector data. There might be some double-counting here (e.g., points that are exact matches for multiple grid cells?
7. For temporal data, we want the names to be dates. For yearly data, you need to impute day and month using `pg_dates()`. See `read_ghsl_population_grid()` in `data_ghsl.R` for an example.
8. Consider renaming `gen_geopko_count()` to `gen_geopko_operations_count()` as there are likely other variables that could be made from geopko data (number of troops, etc). I think `gen_geopko_sum()` is superfluous as you can get these numbers by aggregating `gen_geopko_count` across the years.

Other than that, your code runs well, the sources.csv is properly formatted, the citation is correct, and the documentation reads well.

krihelskog commented:
1. All eurostat scripts are removed from branch. 
2. Space between documentation and functions are removed. 
3. devtools::document() has been run and .rd files are created. 
4. All functions has return(object) in the end of function 
5. Still using terra::rasterize as discussed due to point data. 
6. Checked sum function, it should now be correct. 
7. Added dates column using lubridate package and pg_dates() 
8. Renamed function to gen_geopko_operations_count() and added gen_geopko_troops_count

kvelleby reviewed (approved):
The main remaining issue was how to deal with data that must be aggregated in time. I have added a function `pg_date_intervals()` that produces date intervals based on priogrid global options, and shown how it can be used using the geopko data. Another issue was that the `\insertRef` needs to be wrapped under a `@references` tag in the documentation. I have tested the functions, the documentation, and reference now. GeoPKO is noted with "All rights reserved" as licensing information is not explicitly available, but in email with Lise Hultman and Hanne Fjelde, it was very clear that the data was open.
